Makefile

```diff
@@ -30,6 +30,30 @@ DOCKER_NAME ?= kiali/kiali
 DOCKER_VERSION ?= dev
 DOCKER_TAG = ${DOCKER_NAME}:${DOCKER_VERSION}
 
+# If the IMAGE_PULL_POLICY is not defined and its the short version (eg vX.Y and not vX.Y.Z)
+# then we need to use the image pull policy of 'Always' otherwise the latest update to that
+# branch may not be brought in for users
+ifndef IMAGE_PULL_POLICY
+	SHORT_VERSION=$(shell if [[ ${DOCKER_VERSION} =~ ^v[0-9]+\.[0-9]+$$ ]]; then echo true; fi)
+	ifeq (${SHORT_VERSION}, true)
+		IMAGE_PULL_POLICY = Always
+	endif
+
+	ifeq ("${DOCKER_VERSION}", "latest")
+		IMAGE_PULL_POLICY = Always
+	endif
+
+	ifeq ("${DOCKER_VERSION}", "dev")
+		IMAGE_PULL_POLICY = IfNotPresent
+	endif
```

Comment:
We don't need to check for 'latest' or 'dev' here. Those are the defaults.

Reply:
I wondered about that - the docs indicate the default for "latest" as "Always" is deprecated:

https://kubernetes.io/docs/concepts/configuration/overview/#container-images

`An alternative, but deprecated way to have Kubernetes always pull the image is to use the :latest tag, which will implicitly set the imagePullPolicy to Always.`

so I wonder if we should keep this and explicitly set it to avoid relying on deprecated features?

Reply:
oh, interesting. Ok, I guess this is good then